### PR TITLE
Add REST spec for data access mechanisms

### PR DESCRIPTION
**Pull request by danielcweeks**
This PR add an optional header so that clients can provide information to the REST server related to which data access mechanisms they support.

### CODE REVIEW

File: open-api/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -1453,6 +1456,23 @@ components:
...
+      description: >
+        Optional signal to the server that the client supports data access via
+        a comma-separated list of access mechanisms.  The server may choose to 
+        supply access via any or none of the requested mechanisms.
+      required: false
+      schema:
+        type: string
+        enum:
+          - vended_credentials
+          - remote_signing
```
Comment (rdblue) [inline open-api/rest-catalog-open-api.yaml:1471]: Enums prefer kebab-case to snake_case for consistency throughout.

File: open-api/rest-catalog-open-api.yaml [hunk 2]
```diff
@@ -1453,6 +1456,23 @@ components:
         type: string
       example: "sales"
 
+    data-access:
+      name: X-Iceberg-Data-Access
```
Comment (rdblue) [inline open-api/rest-catalog-open-api.yaml:1460]: This doesn't seem like a clear enough noun to me. How about `X-Supported-Access-Delegation`? I think it should be clear that these are access patterns or methods that are supported by the client. It's also for metadata access as well (manifests and manifest lists). I think having Iceberg in there is nice, but probably not necessary if it is getting too long.

File: open-api/rest-catalog-open-api.yaml (hunk 1 above)
Comment (danielcweeks) [reply open-api/rest-catalog-open-api.yaml]: done

File: open-api/rest-catalog-open-api.yaml (hunk 2 above)
Comment (danielcweeks) [reply open-api/rest-catalog-open-api.yaml]: Updated to `X-Iceberg-Access-Delegation`.  I think it's important to keep `X-Iceberg` as the prefix so there isn't confusion around what the header relates to (pretty standard convention).

**Review by rdblue — APPROVED** [review]
Looks good to me. We may want to also update the documentation to make it clear what these options mean. For example, we could specifically call out the s3, azure, or gcs options that correspond to vended credentials, and link to the remote signing spec for remote signing.

**Pull request merged**